src/modules/identity/RadixIdentity.ts

```diff
@@ -23,18 +23,18 @@
 import RadixSignatureProvider from './RadixSignatureProvider'
 import RadixDecryptionProvider from './RadixDecryptionProvider'
 import RadixAccount from '../account/RadixAccount'
-import { RadixAddress, RadixAtom } from '../atommodel';
+import { RadixAddress, RadixAtom } from '../atommodel'
 
-export default abstract class RadixIdentity implements RadixSignatureProvider, RadixDecryptionProvider {
-    constructor(readonly address: RadixAddress) {
-        this.account = new RadixAccount(address)
-        this.account.enableDecryption(this)
-    }
-
-    public abstract signAtom(atom: RadixAtom): Promise<RadixAtom>
-    public abstract decryptECIESPayload(payload: Buffer): Promise<Buffer>
-    public abstract decryptECIESPayloadWithProtectors(protectors: Buffer[], payload: Buffer): Promise<Buffer>
-    public abstract getPublicKey(): Buffer
-
-    public account: RadixAccount
+export default interface RadixIdentity extends RadixSignatureProvider, RadixDecryptionProvider {
```

Comment:
RadixLedgerIdentity could not directly inherit from RadixIdentity, since initiating the RadixAccount is an asynchronous action (need to get public key from ledger).
When hierarchies start to fall apart, better use composition I think. Hence the refactor...